samples/Balosar/Balosar.Server/Startup.cs

```diff
@@ -0,0 +1,156 @@
+                options.ClaimsIdentity.RoleClaimType = Claims.Role;
+            });
+
+            // Replaces the default schemes to make default [Authorize] attribute work with Tokens.
+            // Alternatively you can use [Authorize(AuthenticationSchemes = OpenIddictValidationAspNetCoreDefaults.AuthenticationScheme)]
+            // in your api methods.
+            services.AddAuthentication(opt =>
+            {
+                opt.DefaultAuthenticateScheme = OpenIddictValidationAspNetCoreDefaults.AuthenticationScheme;
+                opt.DefaultChallengeScheme = OpenIddictValidationAspNetCoreDefaults.AuthenticationScheme;
+                opt.DefaultForbidScheme = OpenIddictValidationAspNetCoreDefaults.AuthenticationScheme;
+            });
+
+            services.AddOpenIddict()
+
+                // Register the OpenIddict core components.
+                .AddCore(options =>
+                 {
+                     // Configure OpenIddict to use the Entity Framework Core stores and models.
+                     // Note: call ReplaceDefaultEntities() to replace the default OpenIddict entities.
+                     options.UseEntityFrameworkCore()
+                            .UseDbContext<ApplicationDbContext>();
+                 })
+
+                // Register the OpenIddict server components.
+                .AddServer(options =>
+                {
+                    options.SetAuthorizationEndpointUris("/connect/authorize");
+                    options.SetLogoutEndpointUris("/connect/logout");
+                    options.SetTokenEndpointUris("/connect/token");
+                    options.SetUserinfoEndpointUris("/connect/userinfo");
+
+                    // Mark the "email", "profile" and "roles" scopes as supported scopes.
+                    options.RegisterScopes(Scopes.Email, Scopes.Profile, Scopes.Roles);
+
+                    // Note: the sample uses authorization code flow but you can enable
+                    // the other flows if you need to support implicit, password or client credentials.                    
+                    options.AllowAuthorizationCodeFlow();
+                    options.AllowRefreshTokenFlow();
+
+                    // Register the signing and encryption credentials.
+                    options.AddDevelopmentEncryptionCertificate()
+                           .AddDevelopmentSigningCertificate();
+
+                    // Register the ASP.NET Core host and configure the ASP.NET Core-specific options.
+                    options.UseAspNetCore()
+                        .EnableAuthorizationEndpointPassthrough()
+                        .EnableLogoutEndpointPassthrough()
+                        .EnableStatusCodePagesIntegration()
+                        .EnableTokenEndpointPassthrough()
+                        .DisableTransportSecurityRequirement(); // During development, you can disable the HTTPS requirement.
```

Comment:
There's a ticket somewhere to enable TLS in all the samples to work around the stricter same-site rules introduced in recent versions of Chrome. Looks like TLS is already enabled in this project, so I suggest we remove this line.